simulation/test_dht.cpp

```diff
@@ -40,14 +40,38 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/session_settings.hpp"
 #include "libtorrent/session.hpp"
 #include "libtorrent/alert_types.hpp"
+#include "libtorrent/ed25519.hpp"
+#include "libtorrent/bencode.hpp"
+#include "libtorrent/kademlia/item.hpp"
+#include <boost/bind.hpp>
 
 namespace lt = libtorrent;
 
+static int g_put_count;
+
+void put_mutable_item_cb(boost::array<char, ed25519_private_key_size> sk
+	, boost::array<char, ed25519_public_key_size> pk
+	, lt::entry& item, boost::array<char, 64>& sig
+	, boost::uint64_t& seq, std::string const& salt)
+{
+	item = "mutable item";
+	seq = 1;
+	std::vector<char> v;
+	lt::bencode(std::back_inserter(v), item);
+	lt::dht::sign_mutable_item(std::make_pair(v.data(), v.size()), std::make_pair(salt.data(), salt.size())
+		, seq, pk.data(), sk.data(), sig.data());
+	g_put_count++;
+}
+
 struct network_config : network_setup_provider
```

Comment:
this class seems to be driving a specific test now. perhaps network_config is not a suitable name anymore. or perhaps the test-specific logic should be factored out into a separate class or function (I'm hoping to add some DHT tests too, and ideally tests should be as small as possible to test a specific feature/property)